framework/db/ActiveQueryTrait.php

```diff
@@ -159,7 +159,10 @@ private function createModels($rows)
      */
     public function findWith($with, &$models)
     {
-        $primaryModel = new $this->modelClass;
+        $primaryModel = reset($models);
+        if (!$primaryModel instanceof ActiveRecordInterface) {
+            $primaryModel = new $this->modelClass;
```

Comment:
exactly. this already helps a lot as you can now use everything except joinWith and asArray. Will address the other issue in another PR.